TestResultSummaryService/Database.js

```diff
@@ -188,8 +228,153 @@ class Database {
+                                                  }
+                                                  }
+                                              },
+                                              0
+                                              ]
+                                          }
+                                          ]
+                                      }
+                                      ]
+                                  }
+                                  ]
+                              },
+                          }
+                      }
+                  },
+            },
+            {
+                $addFields: {
+                    manual_rerun_needed: {
+                        $size:
+                            '$manual_rerun_needed_list'
+                    }
+                  },
+            },
+            {
+                $addFields: {
+                    tests_needed_manual_rerun: {
+                        $reduce: {
+                            input: '$manual_rerun_needed_list',
+                            initialValue: 0,
+                            in: {
+                                $add: [
+                                    '$$value',
+                                    { $ifNull: ['$$this.testSummary.failed', 0] }
+                                ]
+                            }
+                        }
+                    }
+                }
+            },
+            {
+                $addFields: {
+                    manual_rerun_needed_regex: {
+                        $concat: [
+                            '^(',
+                            {
+                                $reduce: {
+                                    input: '$manual_rerun_needed_list',
+                                    initialValue: '',
+                                    in: {
+                                        $cond: [
+                                            { $eq: ['$$value', ''] }, // If the first item
+                                            { $replaceAll: { input: '$$this.buildName', find: '.', replacement: '\\.' } },
+                                            {
+                                                $concat: [
+                                                    '$$value',
+                                                    '|',
+                                                    { $replaceAll: { input: '$$this.buildName', find: '.', replacement: '\\.' } }
+                                                ]
+                                            }
+                                        ]
+                                    }
+                                }
+                            },
+                            ')$'
+                        ]
+                    }
+                }
+            },
+            { $unset: ['childBuilds', 'manual_rerun_needed_list'] }
+        ];
+    }
```

Comment:
This is complex to review and test. Could you write a pipeline query?

Key Performance Improvements
- Avoids multiple $filter operations by pre-filtering failed builds.
- Replaces $reduce with $sum in $lookup, making it more efficient.
- Minimizes $regexMatch usage, reducing processing time.
- Efficient regex generation using $group, avoiding multiple conditionals.